powerdns/tests/test_ownership.py

```diff
@@ -90,3 +101,121 @@ def test_explicit_user(self):
             },
         )
         self.assertOwner(request, 'user2', mailed=True)
+
+
+class TestCreateRecordAccessByServiceOwnership(TestCase):
+
+    def setUp(self):
+        self.clicker = UserFactory(username='clicker')
+        self.some_dude = UserFactory(username='some_dude')
+        self.example_domain = DomainFactory(
+            owner=self.clicker,
+            name='example.com',
+            unrestricted=False,
+            auto_ptr=AutoPtrOptions.NEVER,
+        )
+
+    def _test_create(self, domain_owner, domain_ownership, expected):
```

Comment:
https://github.com/allegro/django-powerdns-dnssec/pull/151/files#diff-c1762345d09453fe3aaa71e8c433604eR75 - test when this `else` block is used (domain has no service)